braindecode/models/tidnet.py

```diff
@@ -0,0 +1,251 @@
+        super().__init__()
+        dim = dim if dim > 0 else dim + 4
+        if dim < 2 or dim > 3:
+            raise ValueError('Only last two dimensions supported')
+        kernel = (filter_len, 1) if dim == 2 else (1, filter_len)
+
+        self.net = nn.Sequential(
+            nn.BatchNorm2d(in_features),
+            activation(),
+            nn.Conv2d(in_features, bottleneck * growth_rate, 1),
+            nn.BatchNorm2d(bottleneck * growth_rate),
+            activation(),
+            nn.Conv2d(bottleneck * growth_rate, growth_rate, kernel,
+                      padding=tuple((k // 2 for k in kernel))),
+            nn.Dropout2d(drop_prob)
+        )
+
+    def forward(self, x):
+        return torch.cat((x, self.net(x)), dim=1)
+
+
+class _DenseSpatialFilter(nn.Module):
+    def __init__(self, in_chans, growth, depth, in_ch=1, bottleneck=4, drop_prob=0.0,
+                 activation=nn.LeakyReLU, collapse=True):
+        super().__init__()
+        self.net = nn.Sequential(*[
+            _DenseFilter(in_ch + growth * d, growth, bottleneck=bottleneck, drop_prob=drop_prob,
+                         activation=activation) for d in range(depth)
+        ])
+        n_filters = in_ch + growth * depth
+        self.collapse = collapse
+        if collapse:
+            self.channel_collapse = _ConvBlock2D(n_filters, n_filters, (in_chans, 1), drop_prob=0)
+
+    def forward(self, x):
+        if len(x.shape) < 4:
+            x = x.unsqueeze(1).permute([0, 1, 3, 2])
+        x = self.net(x)
+        if self.collapse:
+            return self.channel_collapse(x).squeeze(-2)
+        return x
+
+
+class _TemporalFilter(nn.Module):
+    def __init__(self, in_chans, filters, depth, temp_len, drop_prob=0., activation=nn.LeakyReLU,
+                 residual='netwise'):
+        super().__init__()
+        temp_len = temp_len + 1 - temp_len % 2
+        self.residual_style = str(residual)
+        net = list()
+
+        for i in range(depth):
+            dil = depth - i
+            conv = weight_norm(nn.Conv2d(in_chans if i == 0 else filters, filters,
+                                         kernel_size=(1, temp_len), dilation=dil,
+                                         padding=(0, dil * (temp_len - 1) // 2)))
+            net.append(nn.Sequential(
+                conv,
+                activation(),
+                nn.Dropout2d(drop_prob)
+            ))
+        if self.residual_style.lower() == 'netwise':
+            self.net = nn.Sequential(*net)
+            self.residual = nn.Conv2d(in_chans, filters, (1, 1))
+        elif residual.lower() == 'dense':
+            self.net = net
+
+    def forward(self, x):
+        if self.residual_style.lower() == 'netwise':
+            return self.net(x) + self.residual(x)
+        elif self.residual_style.lower() == 'dense':
+            for layer in self.net:
+                x = torch.cat((x, layer(x)), dim=1)
+            return x
+
+
+class _tidnet_features(nn.Module):
```

Comment:
class names should be CamelCase
```suggestion
class _TIDNetFeatures(nn.Module):
```